system/datatypes/mailinglist.php

```diff
@@ -0,0 +1,728 @@
+	 * @param DOMElement $elem
+	 * @param array $data
+	 *
+	 * @return WymHTMLForm
+	 */
+	private function createForm(DOMElement $elem, array $data = []) {
+		$form = new WymHTMLForm($elem);
+		$form->setData($data);
+
+		return $form;
+	}
+
+	/**
+	 * @param string $label
+	 * @param null|string $path
+	 * @param null|string $command
+	 * @param null|string $argument
+	 *
+	 * @return string
+	 */
+	private function createButton($label, $path = null, $command = null, $argument = null) {
+		$path = $this->language . '/' . $this->pagename . ($path ? '/' . $path : '');
+		$_action = makeAction($path, ($command ? $command : ''), ($argument ? $argument : ''));
+		$_button = makeButton(__($label), $_action);
+
+		return $_button;
+	}
+
+	/**
+	 * @param string $selector
+	 *
+	 * @return NodeList
+	 */
+	private function findBySelector($selector) {
+		return $this->html->find($selector);
+	}
+
+	private function redirect($path, $language = null) {
+		header('Location: ' . preserveNotifications($this->generateFullPath($path, $language)));
+		exit;
+	}
+
+	private function reload() {
+		header('Location: ' . preserveNotifications($_SERVER['REQUEST_URI']));
+		exit;
+	}
+
+	private function generateFullPath($path, $language = null) {
+		global $hyphaUrl;
+		$language = null == $language ? $this->language : $language;
+		$path = '' == $path ? '' : '/' . $path;
+
+		return $hyphaUrl . $language . $path;
+	}
+
+	private function isPosted($cmd) {
+		return 'POST' == $_SERVER['REQUEST_METHOD'] && $cmd == $_POST['command'];
+	}
+
+	private function addSpacer(DOMWrap\NodeList $nodeList) {
+		$nodeList->append('<span style="padding: 0 5px;">|</span>');
+	}
+
+	private function send($subject, $message, array $receivers, $from = null) {
+		if (null == $from) {
+			$from = hypha_getEmail();
+		}
+		$messageHtml = '<!DOCTYPE HTML>' . PHP_EOL;
+		$messageHtml .= '<html>' . PHP_EOL;
+		$messageHtml .= '	<head>' . PHP_EOL;
+		$messageHtml .= '		<meta http-equiv=Content-Type content="text/html; charset=utf-8"/>' . PHP_EOL;
+		$messageHtml .= '		<title>' . $subject . '</title>' . PHP_EOL;
+		$messageHtml .= '		<style>' . PHP_EOL;
+		$messageHtml .= '			body {margin-top:10px; margin-left: 10px; font-size:10pt; font-family: Sans; color:black; background-color:white;}' . PHP_EOL;
+		$messageHtml .= '		</style>' . PHP_EOL;
+		$messageHtml .= '	</head>' . PHP_EOL;
+		$messageHtml .= '	<body>' . PHP_EOL;
+		$messageHtml .= $message;
+		$messageHtml .= '	</body>' . PHP_EOL;
+		$messageHtml .= '</html>' . PHP_EOL;
+
+		// send mailing
+		$headers = 'MIME-Version: 1.0' . PHP_EOL;
+		$headers .= 'Content-type: text/html; charset=utf-8' . PHP_EOL;
+		$headers .= 'From: ' . $from . ' <' . $from . '>' . PHP_EOL;
+		$headers .= 'Reply-To: ' . $from . PHP_EOL;
+		$headers .= 'X-Mailer: PHP/' . phpversion();
+
+		foreach ($receivers as $receiver) {
+			mail($receiver, $subject, $messageHtml, $headers . PHP_EOL . 'To: ' . $receiver . PHP_EOL);
+			usleep(rand(500000, 2000000));
+		}
+	}
```

Comment:
If `sendMail()` is going to be used, I think it needs to be unified with the method used in this case.
I'll replace `PHP_EOL` with `\r\n`.

Reply:
Is there anything `sendMail` does not do that this method needs? This method of course generates the HTML content that is mailing-list-specific, so that part would stay in this method, but the generic parts could be handled by `sendMail()`?